### ISSUE TITLE
throw exception when call client.getTransactionReceipt

### ISSUE THREAD

**Issue by Matrix-Zhang** [simolus3/web3dart#99]
the backtrace:

```
Unhandled exception:
type 'List<dynamic>' is not a subtype of type 'List<Map<String, dynamic>>' in type cast
#0      new TransactionReceipt.fromMap (package:web3dart/src/core/transaction_information.dart:151:28)
#1      Web3Client.getTransactionReceipt.<anonymous closure> (package:web3dart/src/core/client.dart:243:53)
#2      _RootZone.runUnary (dart:async/zone.dart:1381:54)
#3      _FutureListener.handleValue (dart:async/future_impl.dart:139:18)
#4      Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:680:45)
#5      Future._propagateToListeners (dart:async/future_impl.dart:709:32)
#6      Future._completeWithValue (dart:async/future_impl.dart:524:5)
#7      _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:32:15)
#8      _completeOnAsyncReturn (dart:async-patch/async_patch.dart:290:13)
#9      Web3Client._makeRPCCall (package:web3dart/src/core/client.dart)
<asynchronous suspension>
#10     Web3Client.getTransactionReceipt (package:web3dart/src/core/client.dart:241:12)
#11     twPointTest (file:///D:/Resource/Codes/learn/dart/transactions/bin/tw_point.dart:53:20)
<asynchronous suspension>
#12     main (file:///D:/Resource/Codes/learn/dart/transactions/bin/main.dart:4:9)
#13     _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:305:32)
#14     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:174:12)

```

**Comment by simolus3** [comment]
Thanks for the report. Can you tell me which `web3dart` version you're using? I assume it's `2.0.0-dev.5`?

**Comment by Marcel2508** [comment]
I'm getting the same error with version 2.0.0-dev.5. Issue seems to be created by the following commit / change: https://github.com/simolus3/web3dart/commit/72111eb81a5d8b1d100758820bb1821034471ca6#diff-169ada7432307607467974d5c0d4b225R152

**Comment by jmank88** [comment]
That change was in response to a similar type error, and the tested cases are actual chain data. Maybe we need to cover more cases. @Matrix-Zhang @Marcel2508 Can you link to a transaction that causes failure? Or provide a code snippet to reproduce?

**Comment by jmank88** [comment]
I'm new to dart, but perhaps just loosening the type from `List<dynamic>` to `List` will fix it. We don't really care about the collection type anyways, since we are casting each element.

**Comment by simolus3** [comment]
I think that change should actually fix this, I just haven't released it yet.

The old `as List<Map<String, dynamic>>` was the problem. I'll release this to pub as soon as [the build](https://travis-ci.com/github/simolus3/web3dart/builds/161619573) goes through.

**Comment by simolus3** [comment]
Released as `2.0.0-dev.6`. Let me know if the problem persists.
